tools/benchcomp/test/test_regression.py

```diff
@@ -737,3 +738,105 @@ def test_command_parser(self):
 
             for item in ["benchmarks", "metrics"]:
                 self.assertIn(item, result)
+
+
+    def test_run_command_visualization(self):
+        """Ensure that the run_command visualization can execute a command"""
+
+        with tempfile.TemporaryDirectory() as tmp:
+            out_file = pathlib.Path(tmp) / str(uuid.uuid4())
+            run_bc = Benchcomp({
+                "variants": {
+                    "v1": {
+                        "config": {
+                            "command_line": "true",
+                            "directory": tmp,
+                        }
+                    },
+                    "v2": {
+                        "config": {
+                            "command_line": "true",
+                            "directory": tmp,
+                        }
+                    }
+                },
+                "run": {
+                    "suites": {
+                        "suite_1": {
+                            "parser": {
+                                "command": """
+                                    echo '{
+                                        "benchmarks": {},
+                                        "metrics": {}
+                                    }'
+                                """
+                            },
+                            "variants": ["v2", "v1"]
+                        }
+                    }
+                },
+                "visualize": [{
+                    "type": "run_command",
+                    "command": f"cat - > {out_file}"
+                }],
+            })
+            run_bc()
+            self.assertEqual(
+                run_bc.proc.returncode, 0, msg=run_bc.stderr)
+
+            with open(out_file) as handle:
+                result = yaml.safe_load(handle)
+
+            for item in ["benchmarks", "metrics"]:
+                self.assertIn(item, result)
+
+
+    def test_run_failing_command_visualization(self):
+        """Ensure that benchcomp terminates normally even when run_command visualization doesn't"""
```

Comment:
Does this mean that benchcomp will still exit with exit status of 0 when a visualization fails?

Reply:
When this particular visualization fails, yes. (There are other visualizations that behave differently, for example `error_on_regression` terminates benchcomp with a return code of 1 if any benchmark regressed).

I don't have a strong opinion about this, I can make benchcomp crash out if this visualization fails, but I think it's sort of a shame to. The visualizations currently run one after another but are supposed to be independent.

Reply:
I would expect that benchcomp exit status reflects whether it succeeds to run all visualizations. You could implement a delayed error approach since there is no dependency on the visualizations.

The `error_on_regression` is not really a visualization, but I don't think we should revisit this. As the name says, the visualization fails if it finds a regression, so I don't see how its behavior is special.

Reply:
Done, now benchcomp terminates with 1 if `run_command` exited with non-zero. It doesn't exit straight away, other visualizations get a chance to run, but the return code at the end gets set to 1.

Reply:
> The `error_on_regression` is not really a visualization, but I don't think we should revisit this. As the name says, the visualization fails if it finds a regression, so I don't see how its behavior is special.

I don't mind renaming it e.g. `post-process` or anything else if you have any suggestions